app/kingdom/knight.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+
+    def __str__(self) -> str:
+        return (f"{self.name} ->\tpwr: {self.power}, "
+                f"hp: {self.hp}, protection: {self.protection}")
+
+    def apply_armour(self, armour: list) -> None:
+        for armour_element in armour:
+            self.protection += armour_element["protection"]
+
+    def apply_weapon(self, weapon: dict) -> None:
+        self.power += weapon["power"]
+
+    def apply_potion(self, effect: dict) -> None:
+        if "power" in effect:
+            self.power += effect["power"]
+        if "protection" in effect:
+            self.protection += effect["protection"]
+        if "hp" in effect:
+            self.hp += effect["hp"]
```

Comment:
Rewrite this code to use iteration and avoid three if conditions here as mentioned [in the checklist](https://github.com/sacrosanctry/py-knights-fighting/blob/develop/checklist.md).